### PR TITLE
rgw: ignore fields placed after "file" in S3's browser uploads.

### PR DESCRIPTION
**Pull request by rzarzynski**
Fixes: http://tracker.ceph.com/issues/17273
Signed-off-by: Radoslaw Zarzynski rzarzynski@mirantis.com

### CODE REVIEW

**Comment by mattbenjamin** [comment]
@rzarzynski this looks still relevant, do you agree?

**Comment by rzarzynski** [comment]
@mattbenjamin: the commit is present also in #11179 as a dependency. I still keep this PR open to allow merging the fix separately (FormPost is quite big).

**Comment by oritwas** [comment]
@rzarzynski please rebase

**Comment by rzarzynski** [comment]
@oritwas: rebased. The conflict was trivial and related to the `override` clean-up.

File: src/rgw/rgw_rest_s3.cc [hunk 1]
```diff
@@ -1909,47 +1909,19 @@ int RGWPostObj_ObjStore_S3::get_policy()
...
   int r = read_data(bl, chunk_size, &boundary, &done);
-  if (r < 0)
+  if (r < 0) {
     return r;
+  }
 
   if (boundary) {
     data_pending = false;
-
-    if (!done) {  /* reached end of data, let's drain the rest of the params */
```
Comment (oritwas) [inline src/rgw/rgw_rest_s3.cc]: I agree we should not pass the arguments but I think we should read all the data

**Comment by rzarzynski** [comment]
@oritwas: just addressed the data draining, rebased to fresh `master` and positively verified with s3-tests.

**Comment by rzarzynski** [comment]
Fixed an obvious thinko. The patch was positively verified together with FormPost-related commits in PR #11179.

**Comment by rzarzynski** [comment]
jenkins test this please

**Comment by rzarzynski** [comment]
jenkins test this please (unrelated test failure)

**Comment by rzarzynski** [comment]
jenkins test this please (due to: `Agent went offline during the build`)

**Comment by oritwas** [comment]
jenkins test this please

**Comment by rzarzynski** [comment]
The commit has been merged as a part of #11179. Closing.

**Pull request closed without merging**